### PR TITLE
feat: support tracing via spandex

### PR DESCRIPTION
**Pull request by zachdaniel**
Resolves: #1122

This is a tool that can provide a lot of insight when trying to diagnose latency, errors, or unexpected behavior in any of the internal applications. Right now the tracing is relatively barebones, but we can add more in depth trace metadata at any point, we just have to know what we're looking for. Better to start with a skeleton for the trace, and fill it in as time goes by.

Left to do:
~I'm going to update `spandex_ecto` to make its truncation take the first `n/2` characters and the last `n/2` characters, instead of the way that it truncates now, which is by taking the first `n` characters. This will help @KronicDeth with his investigations, and allow us to tie queries to postgres logs more easily.~
Instead, we will just include the parameter count as a tag in spandex_ecto

EDIT: this is done.

### Enhancements
* Support tracing. Disabled by default, instructions for enabling are located in the readme.

### CODE REVIEW

**Comment by zachdaniel** [comment]
Still cleaning this up, and also have to backmerge. No point looking yet :)

File: .credo.exs [hunk 1]
```diff
@@ -75,8 +75,9 @@
         # Priority values are: `low, normal, high, higher`
         #
         {Credo.Check.Design.AliasUsage,
-         excluded_namespaces: ~w(Block Blocks Import Socket Task),
-         excluded_lastnames: ~w(Address DateTime Exporter Fetcher Full Instrumenter Monitor Name Number Repo Time Unit),
+         excluded_namespaces: ~w(Block Blocks Import Socket Task SpandexDatadog),
+         excluded_lastnames:
+           ~w(Address DateTime Exporter Fetcher Full Instrumenter Monitor Name Number Repo Time Unit Spec),
```
Comment (sabondano) [inline .credo.exs]: Should we keep these in alphabetical order?

File: README.md [hunk 2]
```diff
@@ -283,6 +283,22 @@ BlockScout is setup to export [Prometheus](https://prometheus.io/) metrics at `/
    3. Click "Load"
 6. View the dashboards.  (You will need to click-around and use BlockScout for the web-related metrics to show up.)
 
+## Tracing
+
+Blockscout supports tracing via
+[Spandex](http://git@github.com:spandex-project/spandex.git). Each application
+has its own tracer, that is configured internally to that application. In order
+to enable it, visit each application's `config/<env>.ex` and update its tracer
+configuration to change `disabled?: true` to `disabled?: false`. Do this for
```
Comment (sabondano) [inline README.md]: 🤔Would it make sense to change these to `enabled: true` to be consistent with how we turn on/off other functionality in: https://github.com/poanetwork/blockscout/blob/d61a06605dd5fdb0c2db2b55809903e9701d0789/apps/explorer/config/config.exs#L18

?

Not a biggy but incase you prefer consistency in this case.

Comment (sabondano) [reply README.md]: ah nvm, didn't realize it was Spandex specific config when I first read this.

File: .credo.exs (hunk 1 above)
Comment (zachdaniel) [reply .credo.exs]: Yeah, I'll change it.

Comment (zachdaniel) [reply .credo.exs]: Wait...I think they already are.

Comment (zachdaniel) [reply .credo.exs]: Nevermind, I was looking at a different branch by accident.

**Pull request merged**